lang/ruby/test/test_logical_types.rb

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'test_help'
+
+class TestLogicalTypes < Test::Unit::TestCase
+  def test_int_date
+    schema = Avro::Schema.parse <<-SCHEMA
+      { "type": "int", "logicalType": "date" }
```

Comment:
I think it would be good to also validate the conversion methods with concrete values that match the ones in [the Java tests](https://github.com/apache/avro/blob/trunk/lang/java/avro/src/test/java/org/apache/avro/data/TestTimeConversions.java). That way we have validation that the dates, times, and timestamps produced by ruby are going to be compatible with the Java library and vice versa.